server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -140,6 +173,10 @@ private static int getTotalShardCount(ClusterState state, Index index) {
         return indexMetadata.getNumberOfShards() * (1 + indexMetadata.getNumberOfReplicas());
     }
 
+    private boolean validateHiddenIndex(String indexName) {
+        return getIgnoreHiddenIndexes() && indexName.charAt(0) == '.';
```

Comment:
Nit - Do we have any definition or disclosure on what happens when an index is created with first character being `.`. If OpenSearch users are creating indices starting with dot, then possible we miss the shard limit validation?

Reply:
Usually indices starting with dot  are created by plugin itself however if users create with dot then yes those indexes will not be validated in the shard limit validation.